vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -191,9 +191,9 @@ def from_guided_params(cls,
 
             if model_with_warn is not None and any_whitespace:
                 logger.info_once(
-                    "%s model detected, consider setting "
+                    f"{model_with_warn} model detected, consider setting "
```

Comment:
shouldn't we fix info_once to support `*args` instead?

f-string with logger is considered not the best practice iirc.

Reply:
I agree, but I think we should patch it first so users don't see errors

Reply:
#17425 cc @hmellor